rust_snuba/src/processors/spans.rs

```diff
@@ -0,0 +1,406 @@
+        let mut tag_keys: Vec<String> = from.tags.clone().into_keys().collect();
+        let mut tag_values: Vec<Value> = from
+            .tags
+            .into_values()
+            .map(|v| match v {
+                Value::String(v) => Value::String(v),
+                Value::Bool(b) => {
+                    let mut v: Vec<char> = b.to_string().as_str().chars().collect();
+                    v[0] = v[0].to_uppercase().nth(0).unwrap();
+                    Value::String(v.into_iter().collect())
+                }
+                _ => Value::String(v.to_string()),
+            })
+            .collect();
+
+        if let Some(http_method) = from.sentry_tags.http_method.clone() {
+            tag_keys.push("http.method".into());
+            tag_values.push(http_method.into());
+        }
+
+        if let Some(status_code) = from.sentry_tags.status_code {
+            tag_keys.push("status_code".into());
+            tag_values.push(status_code.to_string().into());
+        }
+
+        if let Some(transaction_method) = from.sentry_tags.transaction_method.clone() {
+            tag_keys.push("transaction.method".into());
+            tag_values.push(transaction_method.into());
+        }
+
+        Ok(Self {
+            description: from.description,
+            duration: from.duration_ms,
+            end_ms: end_timestamp_ms % 1000,
+            end_timestamp: end_timestamp_ms / 1000,
+            exclusive_time: from.exclusive_time_ms,
+            group: u64::from_str_radix(&from.sentry_tags.group, 16).map_err(|_| InvalidMessage)?,
+            group_raw: from.group_raw,
+            is_segment: from.is_segment,
+            parent_span_id: from.parent_span_id,
+            platform: from.sentry_tags.system.clone(),
+            project_id: from.project_id,
+            retention_days: from.retention_days,
+            segment_id: from.segment_id,
+            segment_name: from.sentry_tags.transaction.clone(),
+            sentry_tag_keys: from.sentry_tags.to_keys(),
+            sentry_tag_values: from.sentry_tags.to_values(),
+            sentry_tags: from.sentry_tags,
+            span_id: from.span_id,
+            span_status: status,
+            start_ms: from.start_timestamp_ms % 1000,
+            start_timestamp: from.start_timestamp_ms / 1000,
+            status,
+            tag_keys,
+            tag_values,
+            trace_id: from.trace_id,
+            transaction_id: from.event_id,
+            transaction_op,
+            ..Default::default()
+        })
+    }
+}
+
+fn hex_to_u64<'de, D>(deserializer: D) -> Result<u64, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let hex = String::deserialize(deserializer)?;
+    u64::from_str_radix(&hex, 16).map_err(serde::de::Error::custom)
+}
+
+#[derive(Clone, Copy, Default, Debug, Deserialize, Serialize)]
+#[serde(rename_all = "snake_case")]
+#[repr(u8)] // size limit in clickhouse
+pub enum SpanStatus {
```

Comment:
I want to get that from `relay` directly but I don't think we can import the crate just yet.

@jan-auer Any ideas on how we could share this?

Reply:
I think it's fine to leave it as is, not a huge fan of adding a relay dependency here